nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/RunNiFi.java

```diff
@@ -1131,6 +1135,22 @@ public void run() {
         this.loggingFutures = futures;
     }
 
+    private Long getPidViaRuntimeMXBean(final Logger logger) {
```

Comment:
Perhaps document that this code is still subject to JVM compatibility? In other words may not work on all JVMs although I can confirm that on most JVMs I've used in the past this approach does work.